src/braket/circuits/circuit.py

```diff
@@ -1270,13 +1270,15 @@ def _generate_frame_wf_defcal_declarations(
                     gate, qubits = key
 
                     # ignoring parametric gates
-                    if hasattr(gate, "angle") and isinstance(gate.angle, FreeParameter):
+                    if hasattr(gate, "parameters") and any(
+                        isinstance(parameter, FreeParameter) for parameter in gate.parameters
+                    ):
                         continue
 
                     gate_name = gate._qasm_name
                     arguments = (
-                        [calibration._format_parameter_ast(gate.angle)]
-                        if hasattr(gate, "angle")
+                        [calibration._format_parameter_ast(value) for value in gate.parameters]
+                        if hasattr(gate, "parameters")
                         else None
```

Comment:
Since the users can pass in their own gate definitions as well, how do we handle those arguments?